pipeline/transform.py

```diff
@@ -0,0 +1,184 @@
+
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize class variables.
+        """
+        self.logger = logger
+
+    def get_data(self) -> dict[tuple]:
+        files = [file for file in glob.glob(
+            f"*.feather*") if os.path.isfile(file)]
+        data = {}
+        for file in files:
+            df = pd.read_feather(file)
+            if "generation" in file:
+                formatted_data = self.generation_transform(df)
+                data['generation'] = formatted_data
+                self.logger.info(
+                    f"""Successfully transformed generation data""")
+            if "demand" in file:
+                formatted_data = self.demand_transform(df)
+                data['demand'] = formatted_data
+                self.logger.info(f"""Successfully transformed demand data""")
+            if "cost" in file:
+                formatted_data = self.cost_transform(df)
+                data['cost'] = formatted_data
+                self.logger.info(f"""Successfully transformed cost data""")
+            if "carbon" in file:
+                formatted_data = self.carbon_transform(df)
+                data['carbon'] = formatted_data
+                self.logger.info(f"""Successfully transformed carbon data""")
+
+        self.delete_read_files(files)
+        return data
+
+    def generation_transform(self, df: pd.DataFrame) -> tuple:
+        df['publish_date'] = df['publishTime'].apply(lambda x: x.split('T')[0])
+        df['gain_loss'] = df['generation'].apply(
+            lambda x: '+' if x > 0 else '-')
+
+        # cutoff = datetime.datetime.now() - datetime.timedelta(hours=1, minutes=5)
+        # df = df.query('publishTime != @cutoff')
+        df = df[['publishTime', 'publish_date', 'fuelType', 'gain_loss',
+                'generation', 'settlementPeriod']]
+        return list(df.itertuples(index=False, name=None))
+
+    def demand_transform(self, df: pd.DataFrame) -> tuple:
+        df = df[['startTime', 'demand']]
+        return list(df.itertuples(index=False, name=None))
+
+    def cost_transform(self, df: pd.DataFrame) -> tuple:
+        df = df[['settlementDate', 'settlementPeriod',
+                 'systemSellPrice', 'systemBuyPrice']]
+        return list(df.itertuples(index=False, name=None))
+
+    def carbon_transform(self, df: pd.DataFrame) -> tuple:
+        new_df = pd.concat(
+            df.apply(self.split_intervals, axis=1).to_list(), ignore_index=True)
+        df = new_df[['from', 'forecast', 'carbon level']]
+        return list(df.itertuples(index=False, name=None))
+
+    def split_intervals(self, row):
+        intervals = pd.date_range(
+            start=row['from'], end=row['to'], freq='5min')
+        return pd.DataFrame({
+            'from': intervals[:-1],
+            'to': intervals[1:],
+            'forecast': row['forecast'],
+            'carbon level': row['carbon level']
+        })
+
+    def delete_read_files(self, files):
```

Comment:
We can delete if we need. I don't really look at the `.feather` files.

However, it just slows us down because the files will get wiped when the lamba shuts down.